core/src/main/java/org/elasticsearch/action/index/TransportIndexAction.java

```diff
@@ -88,6 +90,10 @@ protected void doExecute(final IndexRequest request, final ActionListener<IndexR
         // if we don't have a master, we don't have metadata, that's fine, let it find a master using create index API
         ClusterState state = clusterService.state();
         if (autoCreateIndex.shouldAutoCreate(request.index(), state)) {
+            if (!settings.getAsBoolean(MapperService.INDEX_MAPPER_DYNAMIC_SETTING, MapperService.INDEX_MAPPER_DYNAMIC_DEFAULT)) {
+                String message = String.format("trying to auto create mapping for [%s] in index [%s], but dynamic mapping is disabled", request.type(), request.index());
```

Comment:
This is going to fail the forbidden APIs check for using the default locale; have you been running the `precommit` task or any tasks that depend on `precommit`?